packages/react-native-renderer/src/ReactNativeBridgeEventPlugin.js

```diff
@@ -38,13 +47,21 @@ function accumulateDirectionalDispatches(inst, phase, event) {
       console.error('Dispatching inst must not be null');
     }
   }
-  const listener = listenerAtPhase(inst, event, phase);
-  if (listener) {
+  const listeners = listenersAtPhase(
+    inst,
+    event,
+    phase,
+    event instanceof CustomEvent,
+  );
+  if (listeners && listeners.length > 0) {
     event._dispatchListeners = accumulateInto(
       event._dispatchListeners,
-      listener,
+      listeners,
     );
-    event._dispatchInstances = accumulateInto(event._dispatchInstances, inst);
+    const insts = listeners.map(() => {
+      return inst;
+    });
+    event._dispatchInstances = accumulateInto(event._dispatchInstances, insts);
```

Comment:
Here, too, we create an intermediate array via `map` only to copy it into another array. Let's rewrite this as plainly as possible? If we do the refactoring I suggested earlier then we can push into `event._dispatchInstances` from `pushListenersAtPhase` (in the same place where we're pushing into `event._dispatchListeners`) and we won't need this map. You already have access to `inst` there, too.